### PR TITLE
Tw 619/collection button sell

### PR DESCRIPTION
**Pull request by Boltman92**

### CODE REVIEW

File: src/utils/number.util.ts [hunk 1]
```diff
@@ -67,3 +67,5 @@ const numberWithSpaces = (amount: string) => {
 
   return parts.join('.');
 };
+
+export const valueToDecimals = (value: number, decimals: number) => value / Math.pow(10, decimals);
```
Comment (svjatoslavtt) [inline src/utils/number.util.ts:71]: we have `mutezToTz` util, but I agree that the name is not clear

**Review by svjatoslavtt — APPROVED** [review]

File: src/screens/collection/collection.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+export const Collection = () => {
+  const styles = useCollectionStyles();
+  const selectedAccount = useSelectedAccountSelector();
+  const { params } = useRoute<RouteProp<ScreensParamList, ScreensEnum.Collection>>();
+
+  const collectibles = useCollectibleByCollectionInfo(params.collectionContract, selectedAccount.publicKeyHash);
+
+  const { setInnerScreenIndex } = useInnerScreenProgress(collectibles.length);
+
+  const handleChanged = useCallback((info: { viewableItems: ViewToken[]; changed: ViewToken[] }) => {
```
Comment (herkoss) [inline src/screens/collection/collection.tsx:25]: You don't use `changed` variable, so why do you need it in this function?

File: src/screens/collection/components/collectible-item.tsx [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+              style={[
+                styles.sellButtonText,
+                conditionalStyle(
+                  isOffersExisted && !isOfferFromUser,
+                  styles.sellButtonActive,
+                  styles.sellButtonDisabled
+                )
+              ]}
+            >
+              {isHolder ? (isOffersExisted && !isOfferFromUser ? `Sell for ${highestOffer}` : 'No offers yet') : 'buy'}
```
Comment (herkoss) [inline src/screens/collection/components/collectible-item.tsx:140]: Too hard read this code, let's refactor it

**Review by herkoss — APPROVED** [review]

File: src/screens/collection/components/collectible-item.tsx [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+  const holders = item?.holders?.map(holder => holder.holder_address) ?? [];
+  const isHolder = holders.includes(selectedAccount.publicKeyHash);
+  const isOffersExisted = isDefined(item.highestOffer);
+
+  const handlePress = async () => {
+    const transferParams = isDefined(offer)
+      ? 'fulfill_offer' in offer?.methods
+        ? [offer.methods.fulfill_offer(item.highestOffer?.bigmap_key ?? 1, item.id).toTransferParams()]
+        : [offer.methods.offer_accept(item.highestOffer?.bigmap_key ?? 1).toTransferParams()]
+      : [];
```
Comment (herkoss) [inline src/screens/collection/components/collectible-item.tsx:66]: Don't use one conditional operator inside another, please. It is hard to read this code

File: src/screens/collection/components/collectible-item.tsx [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+              conditionalStyle(isOffersExisted, styles.sellButtonActive, styles.sellButtonDisabled)
+            ]}
+          >
+            <Text
+              style={[
+                styles.sellButtonText,
+                conditionalStyle(isOffersExisted, styles.sellButtonActive, styles.sellButtonDisabled)
+              ]}
+            >
+              {isHolder ? (isOffersExisted ? `Sell for ${highestOffer}` : 'No offers yet') : 'buy'}
```
Comment (herkoss) [inline src/screens/collection/components/collectible-item.tsx:139]: The same

**Review by herkoss — COMMENTED** [review]
Goog job!

File: src/screens/collection/collection.tsx [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+    params.collectionContract,
+    selectedAccount.publicKeyHash,
+    params.type
+  );
+
+  const { setInnerScreenIndex } = useInnerScreenProgress(collectibles.length);
+
+  const handleChanged = useCallback((info: { viewableItems: ViewToken[] }) => {
+    if (isNonEmptyArray(info.viewableItems) && isDefined(info.viewableItems[0].index)) {
+      if (info.viewableItems[0].index === 0) {
```
Comment (herkoss) [inline src/screens/collection/collection.tsx:36]: Magic zero index. What does it mean?

**Review by herkoss — APPROVED** [review]
Well done)

File: src/screens/collection/collection.tsx (hunk 6 above)
Comment (Boltman92) [reply src/screens/collection/collection.tsx:36]: at the initial position, we need to show 1 as the first item in our collection
but we have dummy component in flatlist, for left margin, so we need to figure out numeration this way

**Review by svjatoslavtt — APPROVED** [review]

**Pull request closed without merging**